qiskit/algorithms/gradients/spsa_estimator_gradient.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gradient of Sampler with Finite difference method."""
+
+from __future__ import annotations
+
+from typing import Sequence
+import random
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class SPSAEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation value by the Simultaneous Perturbation Stochastic
+    Approximation (SPSA).
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        epsilon: float = 1e-2,
+        seed: int | None = None,
+        **run_options,
+    ):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            seed: The seed for a random perturbation vector.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting."""
+        self._epsilon = epsilon
+        self._seed = random.seed(seed) if seed else None
```

Comment:
Should we prefer to use a numpy generator here that we can see so its independent. In looking at the BaseEstimator it seems more that's what it does

Reply:
I agree. `np.random.default_rng(seed)` is better to be consistent with other modules in Terra.